zerver/webhooks/freshstatus/view.py

```diff
@@ -0,0 +1,159 @@
+FRESHSTATUS_SERVICES_LIMIT = 5
+
+
+@webhook_view("Freshstatus")
+@has_request_variables
+def api_freshstatus_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    payload: Dict[str, Any] = REQ(argument_type="body"),
+) -> HttpResponse:
+
+    try:
+        body = get_body_for_http_request(payload)
+        subject = get_subject_for_http_request(payload)
+    except KeyError:
+        message = MISCONFIGURED_PAYLOAD_ERROR_MESSAGE.format(
+            bot_name=user_profile.full_name,
+            support_email=FromAddress.SUPPORT,
+        ).strip()
+        send_rate_limited_pm_notification_to_bot_owner(user_profile, user_profile.realm, message)
+
+        return json_error(_("Invalid payload"))
+
+    check_send_webhook_message(request, user_profile, subject, body)
+    return json_success()
+
+
+def get_services_content(services_data: List[Dict[str, Any]]) -> str:
+    services_content = ""
+    for service in services_data[:FRESHSTATUS_SERVICES_LIMIT]:
+        services_content += FRESHSTATUS_SERVICES_ROW_TEMPLATE.format(
+            service_name=service.get("service_name")
+        )
+
+    if len(services_data) > FRESHSTATUS_SERVICES_LIMIT:
+        services_content += FRESHSTATUS_SERVICES_OTHERS_ROW_TEMPLATE.format(
+            services_number=len(services_data) - FRESHSTATUS_SERVICES_LIMIT,
+        )
+    return services_content.rstrip()
+
+
+def get_subject_for_http_request(payload: Dict[str, Any]) -> str:
+    event_data = payload["event_data"]
+    if event_data["event_type"] == "INCIDENT_OPEN" and payload["id"] == "1":
+        return FRESHSTATUS_TOPIC_TEMPLATE_TEST
+    else:
+        return FRESHSTATUS_TOPIC_TEMPLATE.format(title=payload["title"])
+
+
+def get_body_for_http_request(payload: Dict[str, Any]) -> str:
+    event_data = payload["event_data"]
+    if event_data["event_type"] == "INCIDENT_OPEN" and payload["id"] == "1":
+        return get_setup_webhook_message("Freshstatus")
+    elif event_data["event_type"] == "INCIDENT_OPEN":
+        services_data = []
+        for service in payload["affected_services"].split(","):
+            services_data.append({"service_name": service})
+        data = {
+            "title": payload["title"],
+            "description": payload["description"],
+            "start_time": dateutil.parser.parse(payload["start_time"]).strftime(
+                "%d/%m/%Y %H:%M %Z"
+            ),
+            "affected_services": get_services_content(services_data),
+        }
+        return FRESHSTATUS_MESSAGE_TEMPLATE_INCIDENT_OPEN.format(**data)
+    elif event_data["event_type"] == "INCIDENT_NOTE_CREATE":
+        if payload["incident_status"] == "Closed":
+            return FRESHSTATUS_MESSAGE_TEMPLATE_INCIDENT_CLOSED.format(**payload)
+        elif payload["incident_status"] == "Open":
+            return FRESHSTATUS_MESSAGE_TEMPLATE_INCIDENT_NOTE_CREATED.format(**payload)
+    elif event_data["event_type"] == "MAINTENANCE_PLANNED":
```

Comment:
I think this `elif` checks are complex enough that we should move the logic to separate and aptly named functions, such as `get_body_for_maintenance_planned_event`, etc.